weaviate_txtai/ann/weaviate.py

```diff
@@ -0,0 +1,216 @@
+            )
+
+        if self.client.schema.contains(schema):
+            if not self.overwrite_index:
+                raise weaviate.exceptions.ObjectAlreadyExistsException(
+                    f"Index {schema['class']} already exists"
+                )
+            else:
+                self.client.schema.delete_class(schema["class"])
+
+        self.client.schema.create_class(schema)
+        self.index_name = schema["class"]
+
+    def index(self, embeddings):
+        self.append(embeddings)
+
+    def append(self, embeddings):
+
+        with self.client.batch as batch:
+            for embedding in embeddings:
+                random_identifier = uuid.uuid4()
+                object_uuid = generate_uuid5(random_identifier)
+                batch.add_data_object(
+                    data_object={
+                        "docid": self.config["offset"],
+                    },
+                    class_name=self.index_name,
+                    vector=embedding,
+                    uuid=object_uuid,
+                )
+
+                self.config["offset"] += 1
+
+    def _get_uuid_from_docid(self, docid):
+
+        results = (
+            self.client.query.get(self.index_name)
+            .with_additional("id")
+            .with_where(
+                {
+                    "path": ["docid"],
+                    "operator": "Equal",
+                    "valueInt": docid,
+                }
+            )
+            .do()
+        )
+
+        return results["data"]["Get"][self.index_name][0]["_additional"]["id"]
+
+    @check_index_exists
+    def delete(self, ids):
+
+        for id in ids:
+            # TODO: rewrite when weaviate supports IN operator
+            #       See: https://github.com/weaviate/weaviate/issues/2387
+            uuid = self._get_uuid_from_docid(id)
+            self.client.data_object.delete(uuid, class_name=self.index_name)
+
+    @check_index_exists
+    def search(self, queries, limit):
+
+        nearVector = {"vector": queries[0]}
+
+        # use distance to score similarity
+        # TODO: discuss if this is the best way to do it
+        #       in txtai, higher similarity score means more similar
+        #       but in weaviate, lower distance means more similar
+        results = (
+            self.client.query.get(self.index_name, properties=["docid"])
+            .with_additional("distance")
+            .with_near_vector(nearVector)
+            .with_limit(limit)
+            .do()
+        )
+
+        results = results["data"]["Get"][self.index_name]
+
+        return [
+            [(result["docid"], result["_additional"]["distance"]) for result in results]
+        ]
+
+    @check_index_exists
+    def count(self):
+        results = self.client.query.aggregate(self.index_name).with_meta_count().do()
+        return results["data"]["Aggregate"][self.index_name][0]["meta"]["count"]
+
+    def save(self, path):
+
+        logger.warning(
```

Comment:
@davidmezzetti I changed this to a warning instead of an exception so that embeddings.save() can still save the index config, dimensionality reduction model, embedding scoring index, etc. I made a similar change to the load method too.

What do you think?

Reply:
Seems reasonable to me.